### PR TITLE
YouTube now gives 429 errors for CAPTCHAs

### PR DESCRIPTION
**Pull request by Mstrodl**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [ ] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

YouTube seems to give 429s for CAPTCHAs now instead of 402s. It also usually needs the `goojf` cookie in order to download (sometimes after solving the CAPTCHA non-cookie responses will work again, but it's inconsistent). I haven't run into a 402, but I'm leaving it just in case it shows up again.

### CODE REVIEW

**Comment by WoodpeckerBaby** [comment]
tl/dr - every ReCaptcha changes YouTube cookies - you need to port it again - it's the only thing needed plus the same user agent used. 

I would like to add that, based on my own testing, it seems, every time a 429 is presented, and manually a ReCaptcha is solved, Cookies are Changed for YouTube, and for Youtube sites only. It is sufficient to redownload the Cookies from the YouTube homepage and port into youtube-dl. Each individual videos does not require specific cookies. The same user agents used for solving the captcha must also be ported to youtube-dl. 

Aside from that, no additional changes are needed.

**Comment by Mstrodl** [comment]
@WoodpeckerBaby Right, this was a documentation change. The cookie is _set_ upon completing the captcha

**Comment by WoodpeckerBaby** [comment]
Even when a Captcha is solved, it’s frequently represented. It’s both based on rate and amount. 

If you update a lot of files without ”archive.txt” you will get hit with captcha within minutes. 

If you consistently downloading a playlist or a channel, you will get hit with a captcha in hours. 

The key entry in the cookie is the `google abuse exception` line. This line is changed when a captcha is solved. 

So far I can’t seem to find a way to get around that.

**Comment by Mstrodl** [comment]
@WoodpeckerBaby This doesn't seem relevant to this PR which merely suggests that users copy the cookie to work around the 429...
